src/server/hotkey_collector.cpp

```diff
@@ -49,5 +67,51 @@ void hotkey_collector::capture_hash_key(const dsn::blob &hash_key, int64_t weigh
 
 void hotkey_collector::analyse_data() { _internal_collector->analyse_data(); }
 
+void hotkey_collector::on_start_detect(dsn::replication::detect_hotkey_response &resp)
+{
+    auto now_state = _state.load();
+    std::string hint;
+    switch (now_state) {
+    case hotkey_collector_state::COARSE_DETECTING:
+    case hotkey_collector_state::FINE_DETECTING:
+        resp.err = dsn::ERR_SERVICE_ALREADY_EXIST;
```

Comment:
Maybe `ERR_INVALID_STATE` is better, if you think ERR_INVALID_STATE is not suitable to show the error, you can define a new error_code.
The error_code problem is also in other cases in this switch condition.
Besides, I wonder what will rpc client act if it receives this error?